extensions/rest-client/deployment/src/main/java/io/quarkus/restclient/deployment/RestClientProcessor.java

```diff
@@ -421,6 +423,15 @@ void registerProviders(BuildProducer<ReflectiveClassBuildItem> reflectiveClass,
                     .produce(new ReflectiveClassBuildItem(false, false, annotationInstance.value().asClass().toString()));
         }
 
+        // Register @RegisterClientHeaders for reflection
+        for (AnnotationInstance annotationInstance : index.getAnnotations(REGISTER_CLIENT_HEADERS)) {
+            AnnotationValue value = annotationInstance.value();
+            if (value != null) {
+                reflectiveClass
+                        .produce(new ReflectiveClassBuildItem(false, false, annotationInstance.value().asClass().toString()));
```

Comment:
I usually recommend to use `.name().toString()`. The main reason is that for class it's ok but for types the toString() can lead to a non valid class name. Thus I think it's better to get used to always use `name().toString()`.

Reply:
Hm, that's a good point. I just copied the code and did not realize there is `ClassInfo.toString()`. However, for `ClassInfo` it should be safe, in contrast with `org.jboss.jandex.ClassType`...

Reply:
`ClassInfo.toString` is safe, although yeah, calling `.name()` first does remove any possible reason to doubt the code :)